mesonbuild/modules/python.py

```diff
@@ -727,10 +727,27 @@ def find_installation(self, state: 'ModuleState', args: T.Tuple[T.Optional[str]]
         found_modules: T.List[str] = []
         missing_modules: T.List[str] = []
         if python.found() and want_modules:
+            # Python 3.8.x or later require add_dll_directory() to be called on Windows if
+            # the needed modules require external DLLs that are not bundled with the modules.
+            # Simplify things by calling add_dll_directory() on the paths in %PATH%
+            add_paths_cmd = None
+            if hasattr(os, 'add_dll_directory'):
+                paths = os.environ['PATH'].split(os.pathsep)
+                add_paths_cmds = []
+                for path in paths:
+                    if path != '' and os.path.isdir(path):
+                        add_paths_cmds.append("os.add_dll_directory(r'%s')" % path.strip(os.sep))
+                add_paths_cmd = ';'.join(add_paths_cmds)
+
             for mod in want_modules:
-                p, *_ = mesonlib.Popen_safe(
-                    python.command +
-                    ['-c', f'import {mod}'])
+                if add_paths_cmd is not None:
+                    p, *_ = mesonlib.Popen_safe(
+                        python.command +
+                        ['-c', f'import os;{add_paths_cmd};import {mod}'])
+                else:
+                    p, *_ = mesonlib.Popen_safe(
```

Comment:
Rather than having add_paths_cmd be None or a string, why not have it be an empty string and then if there is nothing to add, nothing meaningful gets inserted into the command to run?